### ISSUE TITLE
Provide an option in TSLint for Visual Studio to fail the build if there are Lint Errors

### ISSUE THREAD

**Issue by NoelAbrahams** [aspnet/Tooling#718]
Hi, 

We're trying to use TSLint for Visual Studio 2015 as part of the .Net core Tools Preview 2.

The problem is the linter only runs when **a file is opened**, and then reports any failures as a warning (or optionally as an error) in the error window.

This is just not workable in a solution with hundreds of files.

It would be much nicer if the linter ran as part of the TypeScript build (i.e. MSBuild) and optionally failed the build if there are linting errors.

The errors should be printed to the output window as well to make it easier to fix violations.

Thanks.

**Comment by jakelauer** [comment]
Agreed, this would be extremely helpful for large projects. It is easy to ignore an open file's warnings.
